source/beamline/SGM/energy/SGMEnergyControl.cpp

```diff
@@ -0,0 +1,363 @@
+#include "actions3/AMActionSupport.h"
+#include "SGMGratingTranslationStepControl.h"
+#include "util/AMErrorMonitor.h"
+
+SGMEnergyControl::SGMEnergyControl(SGMUndulatorSupport::UndulatorHarmonic startingUndulatorHarmonic,
+                                   QObject *parent) :
+    AMPseudoMotorControl("Energy", "eV", parent, "SGM Monochromator Energy")
+{
+    energyPositionController_ = 0;
+    startingUndulatorHarmonic_ = startingUndulatorHarmonic;
+
+    gratingAngleControl_ = new AMPVwStatusControl("Grating Angle",
+                                                  "SMTR16114I1002:enc:fbk",
+                                                  "SMTR16114I1002:encTarget",
+                                                  "SMTR16114I1002:status",
+                                                  "SMTR16114I1002:stop",
+                                                  this,
+                                                  5,
+                                                  2.0,
+                                                  new CLSMAXvControlStatusChecker(),
+                                                  1);
+
+	gratingTranslationControl_ = new SGMGratingTranslationStepControl(this);
+
+    undulatorPositionControl_ = new AMPVwStatusControl("Undulator Position",
+                                                       "UND1411-01:gap:mm:fbk",
+                                                       "UND1411-01:gap:mm",
+                                                       "UND1411-01:moveStatus",
+                                                       "SMTR1411-01:stop",
+                                                       this,
+                                                       0.1);
+
+    exitSlitPositionControl_ = new AMPVwStatusControl("Exit Slit Position",
+                                                      "PSL16114I1003:Y:mm:fbk",
+                                                      "PSL16114I1003:Y:mm:encsp",
+                                                      "SMTR16114I1003:status",
+                                                      "SMTR16114I1003:stop",
+                                                      this,
+                                                      0.1,
+                                                      2.0,
+                                                      new AMControlStatusCheckerDefault(1));
+
+    addChildControl(gratingAngleControl_);
+    addChildControl(gratingTranslationControl_);
+    addChildControl(undulatorPositionControl_);
+    addChildControl(exitSlitPositionControl_);
+
+    setMinimumValue(200);
+    setMaximumValue(3000);
+    setTolerance(0.5);
+    setDisplayPrecision(5);
+
+}
+
+bool SGMEnergyControl::shouldMeasure() const
+{
+    return true;
+}
+
+bool SGMEnergyControl::shouldMove() const
+{
+    return true;
+}
+
+bool SGMEnergyControl::shouldStop() const
+{
+    return true;
+}
+
+bool SGMEnergyControl::canMeasure() const
```

Comment:
Do the features available to the child controls impact whether SGMEnergyControl canMeasure()/canMove()/canStop()? Eg. can the energy control stop if the grating angle control can't stop?